CHANGELOG

```diff
@@ -1,3 +1,13 @@
+qtile X.X.X, released XXXX-XX-XX:
+    !!! Config breakage !!!
+	- hooks that pass the qtile object as an argument no longer will. It
+	  can be accessed directly at libqtile.qtile.
+    !!! deprecation warning !!!
+	- defining a main function in your config is deprecated. Instead the
+	  qtile object can be accessed before setup by importing libqtile.qtile
+	  into your config, or after setup by using
+	  hook.subscribe.startup_complete.
```

Comment:
```suggestion
	- defining a main function in your config is deprecated. You should use
	  @hook.subscribe.startup_complete instead. If you need access to the
	  qtile object, import it from libqtile directly.
```

Reply:
Also, it's a config breakage rather than a deprecation now that you removed it from configreader.py, isn't it?

Reply:
I jumped the gun a bit with that one, and misunderstood how the confreader works. You're right. The intended behaviour was:
```
if the config defines main:
    save it as qtile.main, and run it upon start with the deprecation warning
else:
    do nothing with main
````
We don't want `main` in the default config if it's being deprecated, but still want it to be allowed. Maybe we could just add a comment for now in the default config saying it will soon be deprecated, do not use?